runtime/pyproject.toml

```diff
@@ -20,6 +20,7 @@ black = "^24.4.2"
 
 [tool.poetry.scripts]
 qc_runtime = 'runtime.runtime:main'
+qc-report-json = 'runtime.report.json.__main__:main'
```

Comment:
Minor: Perhaps better to use `qc_report_json`, as we have `qc_runtime`, `qc_openscenarioxml`, `qc_opendrive`, `qc_otx`.

Also, it's better to move this `report_json` outside of `runtime`, to make it as an independent module. The structure can be as follows:

```
qc-framework
  -- runtime
  -- report_json
        -- pyproject.toml
        -- report_json
             -- __init__.py
             -- __main__.py
             -- application.py
             -- ...
```

I think we should avoid using `json` as the directory name as it can be confusing with the `json` package from Python.

Reply:
There is a two way strict relation between reporting and actual framework. I'd prefer to keep them together. It is better to avoid to provide multiple modules in a single package, and it makes little sense to me to create an additional package for this project that should be maintained. 

> I think we should avoid using json as the directory name as it can be confusing with the json package from Python.

I prefer avoid repetition of supermodule in submodule name unless strictly necessary. I've difficulties in seeing situations in which the two can be mixed up in code.

Reply:
Hi Matteo, thanks a lot for adding all the 3 report modules. Amazing work :D. It's really cool to see how simple it is to implement all the 3 report modules at once and now with just one `pip install`, users have access to all of them :D.

Regarding the structure, I think it makes sense to group all the report modules into one package, e.g., `asam-qc-report`. But I think we should not include them in the runtime package. Runtime has a different responsibility than the report modules. Also, the structure is not ideal as we have the `report` folder inside `runtime` folder, next to `runtime.py` and `models.py`.

Because we already have the `runtime` package, we can have:
```
qc-framework
  -- runtime
  -- report
        -- pyproject.toml
        -- report_json
        -- report_text
        -- report_github_ci
```

If we see that it's too much effort to maintain two python packages `runtime` and `report`, in the future, we can have one `asam-qc` package and we will have 

```
qc-framework
  -- pyproject.toml
  -- runtime
  -- report
        -- report_json
        -- report_text
        -- report_github_ci
```

Regarding the `json` name, it is just to avoid situation where in the future someone want to write `from report import json`. That `import` will not work or has confusing behavior later in the code. 

Although the above example `import` is also not a good practice neither (should be `from report.json import ...`), but we should avoid builtin names in general if possible.

Reply:
Mh, I'm think we are dealing with an XY Problem ion this discussion. I'll open an issue about that.

Reply:
@hoangtungdinh, I did not found a way to implement (in a clean way) the solutions you proposed. Also the directory `qc-framework` you are referring to does not exist in the current repository.

For me this is a XY problem: more than change the position of `report`, the full package should be configured in such a way it can contains the`runtime` and the other elements required to ship a complete and functioning framework install. Furthermore, there is issue #194 to be addressed, which is really closely related to the actual discussion point.

I'm proposing solution #195, that implements a directory structure similar to the one you proposed but is also _standard_, and tackles the issue of root module names. The final directory structure will be someting like:

```
qc_framework
-- pyproject.tomò
-- qc_framework
   -- runtime
      ...
   -- report
      -- json 
         ...
```

Regarding names for `report` submodules, I have to disagree with your objection of import confusion:

 * the import confusion is simply solved with `from qc_framework.report import json as report_json`, as it is normally done
 * submodule's names have a direct effect on how the user launches the report tools. All the tools are designed to be launched even if **users do not have python script defined in PATH**. Namely you can launch the json report
    * with `qc_report_json`
    * or with `python -m qc_framework.report.json`. This string is exactly the modules names. It makes no sense for users to launch report tools with useless repetitions such as `python -m qc_framework.report.report_json`.

I think users usability takes precedence here.

Finally, regarding packaging report tools in a whole separate package creates a dependency cycle between framework and report tool packages. To use framework you need at least one report tool, report tool can be used only inside the framework application (they need results file as input). Is my opinion it makes little to no sense separate the two when the users cannot have a working installation without both packages. The same concept does not apply to 'qc_baselib`, where a developer can use the models defined inside the base lib to implement its own third party package.

Also, it adds another package to the maintenance list, both for developers and users.

Reply:
Thanks @MatteoRagni for resolving the structure in #195. We can keep the report module names as it is currently too.